src/frontend/js/components/RegisteredCreditCard/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import type * as Joanie from 'types/Joanie';
+
+const messages = defineMessages({
+  inputAriaLabel: {
+    defaultMessage: "{selected, select, true {Unselect} other {Select}} {title}'s card",
+    description: 'ARIA Label read by screen reader to inform which card is focused',
+    id: 'components.RegisteredCreditCard.inputAriaLabel',
+  },
+  expirationDate: {
+    defaultMessage: 'Expiration date: {expirationDate}',
+    description: 'Credit card expiration date label',
+    id: 'components.RegisteredCreditCard.expirationDate',
+  },
+});
+
+export interface Props extends Joanie.CreditCard {
+  selected: boolean;
+  handleSelect: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const RegisteredCreditCard = ({ selected, handleSelect, ...creditCard }: Props) => {
+  const intl = useIntl();
+
+  const expirationDate = useMemo(() => {
+    const { expiration_month: month, expiration_year: year } = creditCard;
+    return intl.formatDate(new Date(year, month - 1, 1), {
+      month: '2-digit',
+      year: 'numeric',
+    });
+  }, [creditCard.expiration_month, creditCard.expiration_year]);
+
+  const inputId = useMemo(() => `registered-credit-card-${creditCard.id}`, [creditCard.id]);
+
+  return (
+    <label className="registered-credit-card">
+      <p className="form-field">
+        <input
+          aria-label={intl.formatMessage(messages.inputAriaLabel, {
+            title: creditCard.title,
+            selected,
+          })}
+          aria-describedby={`credit-card-${creditCard.id}-infos`}
+          checked={selected}
+          className="form-field__checkbox-input"
+          onChange={handleSelect}
+          type="checkbox"
+          id={inputId}
+        />
+        <label htmlFor={inputId} className="form-field__label">
+          <span className="form-field__checkbox-control">
+            <svg role="img" aria-hidden="true">
+              <use href="#icon-check" />
+            </svg>
+          </span>
+        </label>
+      </p>
+      <div id={`credit-card-${creditCard.id}-infos`} className="registered-credit-card__infos">
+        <h6 className="registered-credit-card__name">{creditCard.title || creditCard.brand}</h6>
```

Comment:
As the other things, I'd also say here that maybe using a heading is overkill, since we already have the "Your registered credit card" just before. I'd say a `strong` tag is more fitting.